lib/foreman/cast.rb

```diff
@@ -0,0 +1,22 @@
+module Foreman
+  module Cast
+    def self.to_bool(value)
+      case value
+
+      when String
+        return true if value =~ (/^(true|t|yes|y|1)$/i)
+        return false if value.blank? || value =~ (/^(false|f|no|n|0)$/i)
+
+      when Fixnum
+        return true if value == 1
+        return false if value == 0
+
+      when NilClass
+        return false
+
+      else
+        return value
```

Comment:
Although then we need to look at the usage of the method - in param_true/false we're not expecting an error to be raised, just to find out whether it is true or false, while in another it raised a TypeError, so a Foreman::Exception would be an improvement.